### PR TITLE
MySQL Server Failover Support

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
$subject

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1950

## Examples
```ballerina
Options option = {
    failoverConfig: {
        failoverServers: [
            {
                host: "localhost"
                port: 5506
            },
            {
                host: "localhost",
                port: 3305
            }
        ],
        timeBeforeRetry: 10,
        queriesBeforeRetry: 10,
        failoverReadOnly: false
    }
};
Client dbClient = check new (host, "root", "111", "mydb", 4406, option);
```

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/mysql/Utils.java [hunk 1]
```diff
@@ -32,32 +31,26 @@
...
-            if (socketTimeout > 0) {
-                options.put(Constants.DatabaseProps.SOCKET_TIMEOUT, socketTimeout);
-            }
-            
-            BString serverTimezone = mysqlOptions.getStringValue(Constants.Options.SERVER_TIMEZONE);
-            if (serverTimezone != null) {
-                options.put(Constants.DatabaseProps.SERVER_TIMEZONE, serverTimezone);
-            }
+        long connectTimeout = getTimeout(mysqlOptions.get(Constants.Options.CONNECT_TIMEOUT));
+        if (connectTimeout > 0) {
```
Comment (kalaiyarasiganeshalingam) [inline native/src/main/java/io/ballerina/stdlib/mysql/Utils.java:38]: Don't we need to  send a warning or error message when this value is less than 0?

Comment (niveathika) [reply native/src/main/java/io/ballerina/stdlib/mysql/Utils.java:38]: If it is less than 0, the driver will not timeout.

File: ballerina/client.bal [hunk 2]
```diff
@@ -144,6 +144,7 @@ type ClientConfiguration record {|
 # MySQL database options.
 #
 # + ssl - SSL Configuration to be used
+# + failoverConfig - Server Failover configurations to be used
```
Comment (BuddhiWathsala) [inline ballerina/client.bal:147]:
```suggestion
# + failoverConfig - Server failover configurations to be used
```

**Review by BuddhiWathsala — APPROVED** [review]
LGTM

**Pull request merged**